File: web-server/src/main/java/com/churchofphilippi/webserver/model/News.java
package com.churchofphilippi.webserver.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "news")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class News {

    @Id
    @SequenceGenerator(
            name = "news_sequence",
            sequenceName = "news_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "news_sequence"
    )
    @Column(
            name = "news_id",
            updatable = false
    )
    private Long newsId;

    @Column(
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String content;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dept_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Dept department;

}

File: cop_mobile/cop_mobile/cop_mobile/Model/News.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace cop_mobile.Model
{
    public class News
    {
        public long newsId { get; set; }
        public string content { get; set; }
        public string startDate { get; set; }
        public string endDate { get; set; }
        public Department department { get; set; }
        public Boolean isChurchWide => this.department != null;
        public string title;

        public News(long newsId, string content, string startDate, string endDate, Department department)
        {
            this.newsId = newsId;
            this.content = content;
            this.startDate = startDate;
            this.endDate = endDate;
            this.department = department;
        }

        public string Title
        {
            get
            {
                 if(this.department != null)
                 {
                    return this.department.name;
                 }
                 else
                 {
                    return "";
                 }
            }
        }

    }
}


